api/types/saml.go

```diff
@@ -487,6 +501,12 @@ func (o *SAMLConnectorV2) CheckAndSetDefaults() error {
 			return trace.BadParameter("need roles field in attributes_to_roles")
 		}
 	}
+
+	if o.GetPreferredRequestBinding() != "" {
+		if !slices.Contains([]string{SAMLRequestHTTPRedirectBinding, SAMLRequestHTTPPostBinding}, o.GetPreferredRequestBinding()) {
+			return trace.BadParameter("invalid preferred_request_binding value. It can be either %q or %q", SAMLRequestHTTPPostBinding, SAMLRequestHTTPRedirectBinding)
```

Comment:
We can user array here to make this more scapable when a new value will be added: 
```suggestion
			return trace.BadParameter("invalid preferred_request_binding value. It can be in %q", validSAMLRedirectBindings)
```